src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/Symmetric/SymmetricOneShotBase.cs

```diff
@@ -0,0 +1,383 @@
+            }
+        }
+
+        protected void EncryptOneShot_SpanTest(byte[] plaintext, byte[] ciphertext, PaddingMode padding, CipherMode mode)
+        {
+            using (SymmetricAlgorithm alg = CreateAlgorithm())
+            {
+                alg.Key = Key;
+                byte[] encrypted = mode switch
+                {
+                    CipherMode.ECB => alg.EncryptEcb(plaintext.AsSpan(), padding),
+                    CipherMode.CBC => alg.EncryptCbc(plaintext.AsSpan(), Iv.AsSpan(), padding),
+                    _ => throw new NotImplementedException(),
+                };
+
+                AssertCiphertexts(ciphertext, encrypted, padding, alg.BlockSize / 8);
+            }
+        }
+
+        protected void DecryptOneShot_ArrayTest(byte[] plaintext, byte[] ciphertext, PaddingMode padding, CipherMode mode)
+        {
+            using (SymmetricAlgorithm alg = CreateAlgorithm())
+            {
+                alg.Key = Key;
+                byte[] decrypted = mode switch
+                {
+                    CipherMode.ECB => alg.DecryptEcb(ciphertext, padding),
+                    CipherMode.CBC => alg.DecryptCbc(ciphertext, Iv, padding),
+                    _ => throw new NotImplementedException(),
+                };
+
+                AssertPlaintexts(plaintext, decrypted, padding);
+            }
+        }
+
+        protected void EncryptOneShot_ArrayTest(byte[] plaintext, byte[] ciphertext, PaddingMode padding, CipherMode mode)
+        {
+            using (SymmetricAlgorithm alg = CreateAlgorithm())
+            {
+                alg.Key = Key;
+                byte[] encrypted = mode switch
+                {
+                    CipherMode.ECB => alg.EncryptEcb(plaintext, padding),
+                    CipherMode.CBC => alg.EncryptCbc(plaintext, Iv, padding),
+                    _ => throw new NotImplementedException(),
+                };
+
+                AssertCiphertexts(ciphertext, encrypted, padding, alg.BlockSize / 8);
+            }
+        }
+
+        private static void AssertPlaintexts(byte[] expected, byte[] actual, PaddingMode padding)
+        {
+            if (padding == PaddingMode.Zeros)
+            {
+                Assert.Equal(expected, actual[..expected.Length]);
+                AssertExtensions.FilledWith<byte>(0, actual.AsSpan(actual.Length));
+            }
+            else
+            {
+                Assert.Equal(expected, actual);
+            }
+        }
+
+        private static void AssertCiphertexts(byte[] expected, byte[] actual, PaddingMode padding, int blockSizeBytes)
+        {
+            if (padding == PaddingMode.ISO10126)
+            {
+                // The padding is random, so we can't check the exact ciphertext.
+                Assert.Equal(actual[..^blockSizeBytes], expected[..^blockSizeBytes]);
+            }
+            else
+            {
+                Assert.Equal(actual, expected);
+            }
+        }
+    }
```

Comment:
Consider a "test" to make sure the derived types all specified all the test methods:

https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.IO.Hashing/tests/NonCryptoHashTestDriver.cs#L33-L63